### DISCUSSION TITLE
另一种很新的中文字体网页嵌入方案

### DISCUSSION THREAD

**Discussion by giscus** [ouuan/iles-blog#23] [Comments]
# 另一种很新的中文字体网页嵌入方案

当你看到 一种很新的中文字体网页嵌入方案，但是懒得优化 DP，也不会进化算法，你可以……
整一个另一种很新的中文字体网页嵌入方案（

https://ouuan.moe/post/2023/01/glyph-segregator

**Comment by SunnyCloudYang** [comment]
昨天刚刚看完ayu老师的一种方案，今天就看到了ouuan老师的另一种方案（但我能说都没搞懂怎么用吗www😭）

**Reply by ouuan** [reply]
你的博客好像没用 web font？

**Comment by YfLi03** [comment]
好厉害 orz 

（一直以为 google fonts 在国内完全不能使用，原来还是能访问的）

**Reply by ouuan** [reply]
我说的 Google Fonts 指的是从 Google Fonts 抄来 unicode-ranges 自己划分（

**Comment by ayuusweetfish** [comment]
欧老师牛的🥳 这也太完善了，我那个方案可以退休了ww 以及 segregate 的名字好贴切！

话说全半角引号的问题我记得可以通过 DOM 元素的 lang 属性解决，但是试了一下发现 Noto Serif SC Web Font 可以，本地的 Source Han Serif CN 就不行，感觉是个大坑x（可怕的 Web）

**Reply by ouuan** [reply]
引号的话，我现在是手动给撇号套上一个 span 来改字体，用的比较少所以问题不大（

这篇提到的问题是因为当时的 unicode-range 没包含单引号，霞鹜文楷的引号宽度是可以作为撇号的我就没改字体，现在包含了就没问题了

**Reply by ouuan** [reply]
其实这两个方案不冲突，可以一起用，只是我这样优化了就已经够小了没必要再继续划分 common，如果继续划分的话也可以考虑把设为 common 的概率阈值设成 0.8 之类的

**Reply by ayuusweetfish** [reply]
确实（感觉细分 common 的优化效果本身就有限，访问两三个页面就全加载进来了ww

欧老师这个处理得真的好细致，感觉再简单封装一下就能一键优化任意网站了（之前想着做个更加通用的脚本，但想到可怕的 Web 就望而却步一直咕着x）。赞美×2🎉
